3_run_rsync/tasks/rsync.yml

```diff
@@ -46,9 +46,19 @@
     retries: 100
     delay: 3
 
+  - name: "Ensure directory exists on the source"
+    stat:
+      path: "{{ mig_source_data_location }}"
+    register: source_dir_stat
+
+  - when: not source_dir_stat.stat.exists
+    fail:
+      msg: |
+        The directory {{ mig_source_data_location }} does not exist on the source cluster.
+        Skipping to the next PV...
+
```

Comment:
@pranavgaikwad wonder if we should add this PVs to failed PV list so users can revisit in the next run

Reply:
@alaypatel07 `fail` signal already takes care of that. Check out the new conditional logic on line #81 Basically, any failure inside this `block` triggers it to be called a failed PVC

Reply:
okay makes sense, thanks